patches/0002-Add-crypto-backend-foundation.patch

```diff
@@ -1782,6 +1909,20 @@ index c6a2518f62ff3a..578b4d6f68504c 100644
  	< crypto/rand
  	< crypto/internal/mlkem768
  	< crypto/ed25519
+diff --git a/src/hash/marshal_test.go b/src/hash/marshal_test.go
+index 3091f7a67acede..b066f8281c5265 100644
+--- a/src/hash/marshal_test.go
++++ b/src/hash/marshal_test.go
+@@ -65,6 +65,9 @@ func TestMarshalHash(t *testing.T) {
+ 			}
+ 
+ 			h := tt.new()
++			if _, ok := h.(encoding.BinaryMarshaler); !ok {
```

Comment:
Missing `if boring.Enabled {`? (Just looking at the pattern, haven't checked the context beyond the cutoff. 😄)

Reply:
`boring` is internal to `crypto`, and this test is not under `crypto`. Worth adding the check, anyway, so I've done the usual trick to check if boring is enabled from outside `crypto`.